### PR TITLE
.github: Add Windows CUDA 11.1 workflow

### PR DESCRIPTION
**Pull request by seemethere**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #59961 .github: Enable CUDA 11.1 workflow on prs
* **#59960 .github: Add Windows CUDA 11.1 workflow**

Adds the CUDA 11.1 workflow to GHA

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29116814](https://our.internmc.facebook.com/intern/diff/D29116814)

### CODE REVIEW

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29116814).

File: .circleci/scripts/windows_cuda_install.sh [hunk 1]
```diff
@@ -25,7 +25,7 @@ else
     exit 1
 fi
 
-if [[ "$cuda_major_version" == "11" && "${JOB_EXECUTOR}" == "windows-with-nvidia-gpu" ]]; then
+if [[ "$cuda_major_version" == "11" && "${JOB_EXECUTOR:-}" == "windows-with-nvidia-gpu" ]]; then
```
Comment (seemethere) [inline .circleci/scripts/windows_cuda_install.sh:28]: This script will fail with an unbound error here if JOB_EXECUTOR doesn't exist.

JOB_EXECUTOR isn't defined on GHA so I'm just going to give it a default value of an empty string here.

**Comment by seemethere** [comment]
Workflow successfully executed here: https://github.com/pytorch/pytorch/actions/runs/936701149, just errored out on another DBCPU issue

**Review by janeyx99 — APPROVED** [review]
amazing!!

**Pull request closed without merging**